icu4c/source/common/uresbund.cpp

```diff
@@ -2068,6 +2079,62 @@ ures_getAllItemsWithFallback(const UResourceBundle *bundle, const char *path,
     getAllItemsWithFallback(rb, value, sink, errorCode);
 }
 
+struct GetAllChildrenSink : public ResourceSink {
+
+    // Destination sink
+    ResourceSink& dest;
+    
+    // Original containing resource bundle
+    const UResourceBundle* containingBundle;
+    
+    // Table resource's key path
+    const char* tableKeyPath;
+    
+    GetAllChildrenSink(ResourceSink& dest, const UResourceBundle* containingBundle, const char* tableKeyPath)
+        : dest(dest), containingBundle(containingBundle), tableKeyPath(tableKeyPath) {}
+    virtual ~GetAllChildrenSink();
```

Comment:
Done.  In my defense, I copied this class from the resource sinks in DateTimePatternGenerator, which did neither of these things (anonymous namespace and "override").  Since I'm messing with those sinks here, should I go back and make these changes to them too?

Reply:
I am surprised that the compiler lets you say both `virtual` and `override`... if it works, it's ok.

I might be good to use `override` in the sub-sinks as well, but for now I am happy to get it right in the "API".